warehouse/models/mart/gtfs_quality/fct_monthly_reports_site_organization_gtfs_vendors.sql

```diff
@@ -0,0 +1,104 @@
+{{ config(materialized='table') }}
+
+WITH assessed_entities AS (
+    SELECT *
+    FROM {{ ref('int_gtfs_quality__daily_assessment_candidate_entities') }}
+    -- filter to only organizations that are supposed to be assessed
+    WHERE reports_site_assessed
+),
+
+gtfs_generation_components AS (
+    SELECT *
+    FROM {{ ref('dim_service_components') }}
+    WHERE component_name IN
+        ('GTFS generation', 'Scheduling (Fixed-route)', 'Real-time info',
+        'Arrival predictions', 'GTFS-rt vehicles/trips', 'GTFS Alerts Publication')
+),
+
+dim_orgs AS (
+    SELECT
+        source_record_id, name, organization_type
+    FROM {{ ref('dim_organizations') }}
+    WHERE _is_current
+),
+
+entities_components_joined AS (
+    SELECT DISTINCT
+        assessed_entities.organization_name, assessed_entities.organization_source_record_id,
+        assessed_entities.organization_itp_id,
+        DATE_TRUNC(assessed_entities.date, MONTH) AS date_start,
+        gtfs_generation_components.component_name, gtfs_generation_components.product_name,
+        gtfs_generation_components.product_vendor_organization_source_record_id
+    FROM assessed_entities
+    LEFT JOIN gtfs_generation_components
+        ON assessed_entities.service_key = gtfs_generation_components.service_key
+),
+
+orgs_joined AS (
+    SELECT
+        entities_components_joined.*,
+        dim_orgs.name, dim_orgs.organization_type
+    FROM entities_components_joined
+    LEFT JOIN dim_orgs
+        ON entities_components_joined.product_vendor_organization_source_record_id = dim_orgs.source_record_id
+),
+
+to_aggregate AS (
+    SELECT
+        *,
+        CASE
```

Comment:
Totally optional stylistic point: You could also do this `CASE WHEN` on the initial `gtfs_generation_components` CTE above. My reason to do so would just be that I think that this doesn't rely on any of the joined data so that makes it very clear where it comes from and kind of pre-states "this is what I am going to be using from this table". I think if you do that you could also possibly remove this CTE and thus shorten the query just a bit. 

But it makes no functional difference.